WordPress/gutenberg
# Tests: Adds reporter that shows JS test errors on GitHub when executing GitHub workflows

## Description
Try adding a Jest Reporter that produces an output format for GitHub annotations. 

Based on https://github.com/facebook/jest/pull/11320 (but ported the TypeScript to plain ol' JS since I couldn't get the TS based reporter to work). Quoting that PR:

> Add a Jest reporter to output a format that will be interpreted by GitHub Actions as annotations -- i.e. any errors reported by Jest will show up in a PR's diff below the offending line. This should make for significantly improved Developer Experience, since developers no longer have to scan CI logs for errors, but instead will be notified of them at the location where they happen.
 
This PR is mostly meant to demo that feature; ideally, we'll be able to tweak it and merge it. If https://github.com/facebook/jest/pull/11320 gets merged, we can then consider switching to that, once there's a Jest version that includes it.

## How has this been tested?
See diff view: There should be an annotation below the failing assertion.

To test locally, run
```
GITHUB_ACTIONS=true npm run test-unit -- test/unit/test/index.test.js
```

to see the GHA-specific output format. Confirm that if you omit the `GITHUB_ACTIONS` env var, you won't see that output.

## Screenshots

![image](https://user-images.githubusercontent.com/96308/115601047-9d921200-a2dd-11eb-8284-e7ec37e33366.png)

---

FYI @gziolo
